jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -2413,18 +2411,14 @@ ecma_op_object_own_property_keys (ecma_object_t *obj_p) /**< object */
     {
       ecma_property_t *property_p = prop_iter_p->types + i;
 
-      if (ECMA_PROPERTY_GET_TYPE (*property_p) == ECMA_PROPERTY_TYPE_NAMEDDATA
-          || ECMA_PROPERTY_GET_TYPE (*property_p) == ECMA_PROPERTY_TYPE_NAMEDACCESSOR)
+      if (ECMA_PROPERTY_IS_RAW (*property_p))
       {
         ecma_property_pair_t *prop_pair_p = (ecma_property_pair_t *) prop_iter_p;
 
-        if (ECMA_PROPERTY_GET_NAME_TYPE (*property_p) == ECMA_DIRECT_STRING_MAGIC
-            && prop_pair_p->names_cp[i] >= LIT_NON_INTERNAL_MAGIC_STRING__COUNT
-            && prop_pair_p->names_cp[i] < LIT_MAGIC_STRING__COUNT)
-        {
-          /* Internal properties are never enumerated. */
-          continue;
-        }
+        /* Internal properties are special properties so they must not be present. */
+        JERRY_ASSERT (ECMA_PROPERTY_GET_NAME_TYPE (*property_p) != ECMA_DIRECT_STRING_MAGIC
+                      || prop_pair_p->names_cp[i] < LIT_NON_INTERNAL_MAGIC_STRING__COUNT
+                      || prop_pair_p->names_cp[i] >= LIT_MAGIC_STRING__COUNT);
```

Comment:
This assert hits on debug test262-esnext on 116 tests. Please a take a look what happened here.